third_party/move/move-bytecode-verifier/src/code_unit_verifier.rs

```diff
@@ -43,9 +43,233 @@ impl<'a> CodeUnitVerifier<'a> {
             .map_err(|e| e.finish(Location::Module(module.self_id())))
     }
 
+    fn pattern_check_for_pack(code_opt: Option<&CodeUnit>) -> PartialVMResult<()> {
+        if code_opt.is_none() {
+            return Err(PartialVMError::new(StatusCode::INVALID_STRUCT_API_CODE));
+        }
+        let code = code_opt.unwrap();
+
+        if code.code.len() < 2 {
+            return Err(PartialVMError::new(StatusCode::INVALID_STRUCT_API_CODE));
+        }
+
+        match (
+            &code.code[code.code.len() - 1],
+            &code.code[code.code.len() - 2],
+        ) {
+            (
+                &Bytecode::Ret,
+                &Bytecode::Pack(_)
+                | &Bytecode::PackGeneric(_)
+                | &Bytecode::PackVariant(_)
```

Comment:
Is it okay not to match the function signature with the variant being packed in the bytecode?